lib/services/permissions.service.js

```diff
@@ -1,31 +1,22 @@
-const NodeCache = require('node-cache');
 const config = require('../config');
-const PermissionError = require('../errors/permission.error');
-const { SOURCE_INVALID, SOURCE_MISSING } = require('../errors/error.messages');
 const datasources = require('./datasources');
+const { getCache, setCache } = require('./cache.service');
 
-const permissionCache = new NodeCache({ stdTTL: 600, checkperiod: 60 });
-
-async function getPermissions(token, requestedsource) {
-  const configuration = config.getConfig();
-  const source = requestedsource || configuration.source;
-
-  if (!source) throw new PermissionError(SOURCE_MISSING);
-  if (!datasources[source]) throw new PermissionError(SOURCE_INVALID);
-
+async function getPermissions(token, source) {
+  const { sources } = config.getConfig();
   const cacheKey = `${source}.${token}`;
-  const cachedPermissions = permissionCache.get(cacheKey);
-
-  if (cachedPermissions && configuration.cache) {
-    return cachedPermissions;
-  }
 
-  const permissions = datasources[source].getPermissions(token);
+  let permissions = getCache(cacheKey);
 
-  if (configuration.cache) {
-    permissionCache.set(cacheKey, permissions);
+  if (!permissions) {
+    if (typeof sources[source] === 'function') {
+      permissions = sources[source](token);
```

Comment:
Moet dit geen **async** functies toelaten?

Reply:
Deze worden toegelaten. de voorgedefinieerde functies zijn ook async.
de functie word resolved in de permissionvalidation.service
Het is misschien wel leesbaarder op deze manier en duidelijker om effectieve waarden te cachen ipv promises
aangepast.